flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -113,4 +117,51 @@ private ExternalResourceUtils() {
 
 		return externalResourceConfigs;
 	}
+
+	/**
+	 * Instantiate the {@link ExternalResourceDriver}s for all of enabled external resources. {@link ExternalResourceDriver}s
+	 * are mapped by its resource name.
+	 */
+	public static Map<String, ExternalResourceDriver> externalResourceDriversFromConfig(Configuration config, PluginManager pluginManager) throws Exception {
+		final Set<String> resourceSet = getExternalResourceSet(config);
+		LOG.info("Enabled external resources: {}", resourceSet);
+
+		if (resourceSet.isEmpty()) {
+			return Collections.emptyMap();
+		}
+
+		final Iterator<ExternalResourceDriverFactory> factoryIterator = pluginManager.load(ExternalResourceDriverFactory.class);
+		final Map<String, ExternalResourceDriverFactory> externalResourceFactories = new HashMap<>();
+		factoryIterator.forEachRemaining(
+			externalResourceDriverFactory -> {
+				externalResourceFactories.put(externalResourceDriverFactory.getClass().getName(), externalResourceDriverFactory);
+			});
```

Comment:
Should we load all the driver factories?
What if for each configured resource name and factory, we call plugin manager to load that driver class? I think that should avoid loading unnecessary plugins.
Moreover, if there are multiple plugins with the configured class names, we should load and use only one of them.